### PR TITLE
Gradle upgrade fixes

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
For #4861 

- build JVM from 8 to 11
- sourceCompatibility from 1_6 to 1_7
- updated deprecated gradle language
- Satisfies :android:lintDebug errors
- Tries to prevent it crashing all the time by matching kotlin-reflect to main kotlin version (1.5.20 soon please)
- Some linting

*** Still fails ***

Mix and match as you wish.

### CODE REVIEW

**Comment by SomeTroglodyte** [comment]
Oh, you reverted. Well, some of the changes in here might still help in the long run and not hurt.

Primary lesson _I_ learned: Use terminal, `export JAVA_HOME = <some openjdk 11 or whatever matches github's VM>`, then ./gradlew clean --warning-mode all` and `./gradlew check --warning-mode all`... That, and Studio's inspections actually help you nicely in these build scripts.

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -3,5 +3,5 @@ plugins {
 }
 
 repositories {
-    jcenter()
+    mavenCentral()  // "JFrog announced JCenter's shutdown in February 2021"
```
Comment (yairm210) [inline buildSrc/build.gradle.kts:6]: Gosh

**Comment by yairm210** [comment]
I think the changes here are good changes in any case.
It's better to make these changes before upgrading the gradle version to work incrementally :)

**Comment by SomeTroglodyte** [comment]
Hours of trial and error, hoping to save _you_ a portion of that time. Adopting - would be part 'yes please why not', part 'would need to check whether that's not introduced until gradle 7', and part political - like what exactly would suffer if we dropped sourceCompatibility = JavaVersion.VERSION_1_6.

I haven't actually _tested_ the syntactical changes with gradle 6, but would guess good chances.

Note on lint - that Android lint: disable("all") _does_ what it sounds like, but in my tests I _still_ got crash dumps from the linting process. Since I fixed that "MissingTranslation" I went back to default - and that `isAbortOnError = false` is default, too. But, as said, the config in this PR runs fine locally but not as github action - at least not on my fork.

**Comment by yairm210** [comment]
A. On upgrading to sourceCompatibility 1.7, that's an inevitability of time. If it does exclude some users, they're unsupportable almost by definition

B. We can start with small changes and see what works, not everything needs to be in the same pr

C. Any change that breaks the github actions breaks all tests and deployments, so that's obviously a no-go

**Comment by yairm210** [comment]
All implemented and running :)

**Pull request closed without merging**